cadccutout/cadccutout/utils.py

```diff
@@ -1,24 +1,112 @@
+from astropy.io.fits import Header
+from astropy.wcs import WCS
 
-__all__ = ['to_num', 'is_integer', 'is_string']
+__all__ = ['to_num', 'is_integer', 'is_string', 'to_astropy_header',
+           'to_astropy_wcs']
 
 
 def to_num(s):
+    '''
+    Convert the given value to an integer, if applicable, or a float.
+    '''
     try:
         return int(s)
     except ValueError:
         return float(s)
 
 
 def is_string(s):
+    '''
+    Determine if the given argument is a String.
+    '''
     return s == str(s)
 
 
 def is_integer(s):
-    if isinstance(s, tuple):
-        return False
-
+    '''
+    Determine if the given argument is an Integer.
+    '''
     try:
         int(s)
         return True
     except ValueError:
         return False
+
+
+def to_astropy_header(header_dict, decompress=True):
+    '''
+    Build an AstroPy header instance, filtering out empty header cards or
+    empty values.
+
+    This function will support a fitsio FITSHDR object or a regular dictionary.
+
+    This assumes a FITS file.
+    '''
+
+    comment_keyword = 'COMMENT'
+    history_keyword = 'HISTORY'
+
+    astropy_header = Header()
+
+    for key in header_dict:
+        if key:
+            value = header_dict[key]
+
+            if value:
+                if key == comment_keyword:
+                    str_value = str(value).replace('\n', '  ')
+                    astropy_header.add_comment(str_value)
+                elif key == history_keyword:
+                    str_value = str(value).replace('\n', '  ')
+                    astropy_header.add_history(str_value)
+                else:
+                    astropy_header[key] = value
+
+    # Set the uncompressed NAXIS values to those of the compressed types.
+    if decompress and 'ZNAXIS' in astropy_header:
+        filtered_headers = filter(lambda key: 'ZNAXIS' in key, astropy_header)
```

Comment:
I had a bit of a tough time understanding this block. Part of it is because I'm not very familiar with the `filter` function - thanks for exposure. Could you rename `filtered_headers` to `znaxis_headers`? It makes more sense to me.

Reply:
Done, thanks.